src/Raven.Server/RavenServer.cs

```diff
@@ -344,6 +345,52 @@ void ConfigureKestrel(KestrelServerOptions options)
             }
         }
 
+        private void UpdateCertificateExpirationAlert(object sender = null, EventArgs args = null)
```

Comment:
@iftahbe maybe let's just invoke this inside `OnServerCertificateChanged` ? Instead of having two separate callbacks we will have one? You do not need `sender` and `args` then also and you do not have to deal with removing the event because we are handling that already for `OnServerCertificateChanged`.

Reply:
Probably worth wrapping also everything in try catch and logging the error.